provider/pkg/e2e_test.go

```diff
@@ -15,7 +15,41 @@ import (
 )
 
 func TestApi(t *testing.T) {
-	e2eTest("api").Run(t)
+	e2eTest("api", providertest.WithBaselineVersion("1.0.1")).Run(t)
+}
+
+func TestBinaryMediaTypesNoInput(t *testing.T) {
+	e2eTest("binaryMediaTypesNoInput", providertest.WithProviderName("aws-apigateway"),
+		providertest.WithBaselineVersion("2.1.0"),
+		providertest.WithSkippedUpgradeTestMode(
+			providertest.UpgradeTestMode_Quick,
+			"Quick mode is only supported for providers written in Go at the moment")).Run(t)
+}
+
+func TestSimpleNoBinaryMediaType(t *testing.T) {
+	e2eTest("simple").Run(t)
+}
+
+func TestSimpleWithBinaryMediaType(t *testing.T) {
```

Comment:
Do these tests fail before the changes to the provider? Or do we need to find a way to write some asserts?

Reply:
Yeah, good point we should probably write tests for the actual bug fixed here.

Reply:
Yeah that would be ideal. If the tests still pass without the fix then if we silently regress we will not notice.